src/Interfaces/AuthenticationProviderInterface.php

```diff
@@ -0,0 +1,11 @@
+<?php
+declare(strict_types=1);
+
+namespace Firehed\API\Interfaces;
+
+use Psr\Http\Message\ServerRequestInterface;
+
+interface AuthenticationProviderInterface
+{
+    public function authenticate(ServerRequestInterface $request): AuthenticationContainerInterface;
```

Comment:
How to indicate no authentication was performed (or successful) is currently UB. At the very least, a doc comment should be added.

From a fail-closed perspective, I'm thinking it should throw a (new) `AuthenticationException` if auth fails in any way - malformed or missing headers, signature verification failing, etc. If an endpoint is requesting authn, there's no use case for returning a not-authn result.

This means that, beyond documentation, an `AuthenticationProviderInterfaceTestTrait` should be added for success and fail cases, with the latter asserting an appropriate exception is thrown (An abstract dataprovider should suffice)